File: app/src/main/java/rusha/x/basket/BasketListAdapter.java
package rusha.x.basket;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

import rusha.x.databinding.BasketItemBinding;

public class BasketListAdapter extends RecyclerView.Adapter<BasketListAdapter.ItemViewHolder> {

    private BasketViewModel viewModel;
    public BasketListAdapter(BasketViewModel viewModel) {
        this.viewModel = viewModel;
    }

    private List<BasketItem> itemsToAdopt = new ArrayList<>();

    public void setItemsToAdopt(List<BasketItem> items) {
        itemsToAdopt = items;
        notifyDataSetChanged();
    }

    @Override
    public int getItemCount() {
        return itemsToAdopt.size();
    }

    @Override
    public ItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType){
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        BasketItemBinding binding = BasketItemBinding.inflate(inflater, parent, false);
        return new ItemViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(BasketListAdapter.ItemViewHolder holder, int position) {
        BasketItem itemOnPosition = itemsToAdopt.get(position);
        holder.bind(itemOnPosition);
    }

    class ItemViewHolder extends RecyclerView.ViewHolder {
        public void bind(BasketItem item) {
            binding.getRoot().setOnClickListener ( v -> {
                viewModel.onBasketItemClick(item);
            });
            binding.basketItemView.setText(item.product.name);
            binding.basketCountView.setText(String.valueOf(item.count));
        }
        private BasketItemBinding binding;
        public ItemViewHolder(BasketItemBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }
}

File: app/src/main/java/rusha/x/basket/BasketListAdapter.kt
package rusha.x.basket

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.basket_item.view.*
import rusha.x.R

class BasketListAdapter(
    private val viewModel: BasketViewModel
) : RecyclerView.Adapter<BasketListAdapter.ItemViewHolder>() {

    var itemsToAdopt: List<Basket.Item> = listOf()
    override fun getItemCount(): Int {
        return itemsToAdopt.size
    }

    override fun onCreateViewHolder(
        parent: ViewGroup, viewType: Int
    ): BasketListAdapter.ItemViewHolder {
        val inflater = LayoutInflater.from(parent.context) // была опечатка. LayoutInfla-y-ter
        val view = inflater.inflate(
            R.layout.basket_item,
            parent,
            false
        )
        return ItemViewHolder(containerView = view)
    }

    override fun onBindViewHolder(holder: BasketListAdapter.ItemViewHolder, position: Int) {
        val itemOnPosition = itemsToAdopt.get(index = position)
        holder.bind(item = itemOnPosition)
    }

    inner class ItemViewHolder(
        val containerView: View
    ) : RecyclerView.ViewHolder(containerView) {
        fun bind(item: Basket.Item) {
            containerView.setOnClickListener {
                viewModel.onBasketItemClick(item)
            }
            containerView.basketItemView.text = item.product.name
            containerView.basketCountView.text = item.count.toString()
        }
    }
}